src/main/java/com/xjeffrose/chicago/db/StorageProvider.java

```diff
@@ -21,4 +24,11 @@
   void close();
 
   void open();
-}
\ No newline at end of file
+
+  void setZkClient(ZkClient zkClient);
```

Comment:
This should only be the case if you're trying to call methods on the zkClient before it's been started, it should be reasonable for you to pass the zkClient in the constructor and ONLY have the constructor store the zkClient. Then later once the zkClient has been started you can start the class that is using the zkClient and it can call those methods.